pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -628,6 +594,61 @@ func filterRemaining(kvs []streampb.StreamEvent_KV) []streampb.StreamEvent_KV {
 	return remaining[:j]
 }
 
+func (lrw *logicalReplicationWriterProcessor) setupBatchHandlers(ctx context.Context) error {
```

Comment:
hrm. asid from the kv leasing bug, why is this change useful? We should strive for distsql replanning to be really cheap.

Reply:
I'm not sure I follow the link to leasing but as far as replans:

> We should strive for distsql replanning to be really cheap

I donno. I think there are three things here: 1) reading an int from a setting and recreating a slice if its length does not match is always going to be cheaper than redoinging _any_ lost work by starting whole are processors 2) there will always be lost work, because we don't want to spend all our time checkpointing when we do work and will have some some amount of uncheckpointed work if we want throughput 3) a total pause/resume has to lose _a lot of work_, compared to a replan by a continuing coordinator, since it has to rely on only _persisted_ progress and we never want to write job progress more than once or twice a minute (ideally even less really).

Additionally, for the purposes of running experiments, it also _replans_ which means the before and after have a lot more than just the worker count altered.

Another thing we strive for is for settings taking effect when changed -- they are our _dynamic_ configuration tool, compared to cli flags or env vars -- specifically intended to take as close to immediate effect as possible without you needing to restart things to apply them.

Reply:
Alright i'm convinced. Carry on!